### PR TITLE
Migrate to centralized GitHub workflows

### PR DESCRIPTION
**Pull request by nicholasmhughes**
### What does this PR do?
This PR implements calls to "reusable workflows" housed in the `salt-extensions/central-artifacts` repository. This should allow us to more easily keep GitHub Actions up-to-date without requiring PRs to be opened for every extension. The proposed workflow YAML can be viewed in the `central-artifacts` repository, but for posterity the Push/PR and Release pipelines look something like this:

* **Push/PR**
  * Automatic Processes (kicked off by push/PR)
    * Run pre-commit checks
    * Run test matrix for Linux, Windows, and macOS
    * Run doc build
* **Tag Creation**
  * Manual Processes
    * Create a PR to update the CHANGELOG with `towncrier`
    * Create a tag at the merged commit above for the release version using semantic versioning (i.e. `v1.0.0`)
  * Automatic Processes (kicked off by tag creation)
    * Run pre-commit checks
    * Run test matrix for Linux, Windows, and macOS
    * Run doc build
    * Build the package
    * Release the package to Test PyPi (uses the saltext-community account)
    * Create a GitHub Release for the package
    * Release the package to PyPi (uses the saltext-community account)

### Commits signed with GPG?
Yes

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

File: .github/workflows/test-action.yml [hunk 1]
```diff
@@ -1,427 +0,0 @@
...
-          - python-version: '3.10'
-            salt-version: '3006.4'
-
-    steps:
-    - uses: actions/checkout@v4
-
-    - uses: eLco/setup-vault@v1.0.2
-      with:
-        vault_version: 1.15.4
-
```
Comment (lkubb) [inline .github/workflows/test-action.yml:36]: **question (blocking)**: How can we make sure this runs as part of the workflow when using central artifacts? Without Vault installed, most of the test suite will be skipped.

**Review by lkubb — CHANGES_REQUESTED** [review]
Thanks for doing all this work, very happy about more centralization in the `salt-extensions` org!

Do you have a solution handy for the Vault setup or should I look into it?

Also, the `pre-commit` workflow is failing, maybe this repo's secrets need to be updated (?).

**Comment by nicholasmhughes** [comment]
I probably should've put this in Draft status... but here we are. Pre-commit is failing due to a problem with getting changed files via Action. I think it's because I indiscriminately updated the action version, and there's probably an output format change... Looking into it later today.

File: .github/workflows/test-action.yml (hunk 1 above)
Comment (nicholasmhughes) [reply .github/workflows/test-action.yml:36]: For now, I've added a flag for setting up Vault before the tests run. We'll roll with that for a while and see if we get too much sprawl from other extensions requirements.

**Comment by nicholasmhughes** [comment]
Ok. All fixed up. Let me know if you think this will work for our needs.

**Review by lkubb — APPROVED** [review]
Love it, thank you very much! 👍 Now we're just missing readthedocs for `salt-extensions`, then everything should be neat.

I will make this enhancement an optional part of the copier template as well.

Sorry for taking so long to review btw, somehow missed the update. 

Have a happy new year! 🚀

**Pull request merged**